File: Java/Fila.java

package com.mycompany.listamain;


public class Fila extends Lista {
    @Override
    public void retirar() {
        if (tamanho() > 0) {
            lista.remove(0);
        } else {
            System.out.println("ERRO, Lista vazia");
        }
    }
}

File: Rust/fila.rs
use crate::Lista;

pub struct Fila {
    pub lista: Lista,
}

impl Fila {
    pub fn new() -> Self {
        Fila { lista: Lista::new() }
    }

    pub fn retirar(&mut self) {
        if self.lista.tamanho() > 0 {
            self.lista.retirar();
        } else {
            println!("ERRO, Lista vazia");
        }
    }
}
